### PR TITLE
Move bulk API's batch_size parameter to processors

### PR DESCRIPTION
**Pull request by chishui**
### Description
1. Mark `batch_size` parameter of bulk API to deprecated
2. Change `batch_size` parameter's default value from 1 to 2147483647
3. Add  `batch_size` to `text_embedding` and `sparse_encoding` processors

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7717

### Version
2.16

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @chishui!! Is this ready for a doc review?

**Comment by chishui** [comment]
@hdhalter I'll post a comment here once this PR is merged and the doc is ready for review, thanks!

**Comment by chishui** [comment]
@hdhalter @Naarcha-AWS  The PR this documentation depending on has been merged

File: _api-reference/document-apis/bulk.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ routing | String | Routes the request to the specified shard.
...
-batch_size | Integer | Specifies the number of documents to be batched and sent to an ingest pipeline to be processed together. Default is `1` (documents are ingested by an ingest pipeline one at a time). If the bulk request doesn't explicitly specify an ingest pipeline or the index doesn't have a default ingest pipeline, then this parameter is ignored. Only documents with `create`, `index`, or `update` actions can be grouped into batches.
+batch_size | Integer | (Deprecated) Specifies the number of documents to be batched and sent to an ingest pipeline to be processed together. Default is `2147483647` (documents are ingested by an ingest pipeline all at once). If the bulk request doesn't explicitly specify an ingest pipeline or the index doesn't have a default ingest pipeline, then this parameter is ignored. Only documents with `create`, `index`, or `update` actions can be grouped into batches.
```
Comment (Naarcha-AWS) [inline _api-reference/document-apis/bulk.md:62]:
```suggestion
batch_size | Integer | **(Deprecated)** Specifies the number of documents to be batched and sent to an ingest pipeline to be processed together. Default is `2147483647` (documents are ingested by an ingest pipeline all at once). If the bulk request doesn't explicitly specify an ingest pipeline or the index doesn't have a default ingest pipeline, then this parameter is ignored. Only documents with `create`, `index`, or `update` actions can be grouped into batches.
```

File: _ingest-pipelines/processors/sparse-encoding.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ The following table lists the required and optional parameters for the `sparse_e
 `field_map.<vector_field>`  | String | Required | The name of the vector field in which to store the generated vector embeddings.
 `description`  | String | Optional  | A brief description of the processor.  |
 `tag` | String | Optional | An identifier tag for the processor. Useful for debugging to distinguish between processors of the same type. |
+`batch_size` | Integer | Optional | Specifies the number of documents to be batched and processed each time. Default value is 1. |
```
Comment (Naarcha-AWS) [inline _ingest-pipelines/processors/sparse-encoding.md:44]:
```suggestion
`batch_size` | Integer | Optional | Specifies the number of documents to be batched and processed each time. Default is `1`. |
```

File: _ingest-pipelines/processors/text-embedding.md [hunk 3]
```diff
@@ -41,6 +41,7 @@ The following table lists the required and optional parameters for the `text_emb
 `field_map.<vector_field>`  | String | Required | The name of the vector field in which to store the generated text embeddings.
 `description`  | String | Optional  | A brief description of the processor.  |
 `tag` | String | Optional | An identifier tag for the processor. Useful for debugging to distinguish between processors of the same type. |
+`batch_size` | Integer | Optional | Specifies the number of documents to be batched and processed each time. Default value is 1. |
```
Comment (Naarcha-AWS) [inline _ingest-pipelines/processors/text-embedding.md:44]:
```suggestion
`batch_size` | Integer | Optional | Specifies the number of documents to be batched and processed each time. Default is `1`. |
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _ml-commons-plugin/remote-models/batch-ingestion.md [hunk 4]
```diff
@@ -14,10 +14,11 @@ grand_parent: Integrating ML models
 
 If you are ingesting multiple documents and generating embeddings by invoking an externally hosted model, you can use batch ingestion to improve performance.
 
-The [Bulk API]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) accepts a `batch_size` parameter that specifies to process documents in batches of a specified size. Processors that support batch ingestion will send each batch of documents to an externally hosted model in a single request.
+When using [Bulk API]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) to ingest documents, processors that support batch ingestion will split documents into batches and send each batch of documents to an externally hosted model in a single request.
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/batch-ingestion.md:17]:
```suggestion
When using the [Bulk API]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) to ingest documents, processors that support batch ingestion will split documents into batches and send each batch of documents to an externally hosted model in a single request.
```

File: _ml-commons-plugin/remote-models/batch-ingestion.md [hunk 5]
```diff
@@ -222,7 +224,7 @@ PUT /_ingest/pipeline/nlp-ingest-pipeline
 
 ## Step 6: Perform bulk indexing
 
-To ingest documents in bulk, call the Bulk API and provide the `batch_size` and `pipeline` parameters. If you don't provide a `pipeline` parameter, the default ingest pipeline for the index will be used for ingestion:
+To ingest documents in bulk, call the Bulk API and provide the `pipeline` parameter. If you don't provide a `pipeline` parameter, the default ingest pipeline for the index will be used for ingestion:
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/batch-ingestion.md:227]:
```suggestion
To ingest documents in bulk, call the Bulk API and provide the `pipeline` parameter. If you don't provide a `pipeline` parameter, then the default ingest pipeline for the index will be used for ingestion:
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS @chishui Just a couple of minor additions. Thanks!

**Pull request merged**